pennylane/ops/identity.py

```diff
@@ -281,7 +281,7 @@ def circuit():
 
     """
 
-    grad_method = "A"
+    grad_method = None
```

Comment:
May I suggest to put `grad_method="F"` here? Setting it to `None` has the disadvantage that `param_shift` will think it can "tackle" a global phase with `return qml.state()`, because it thinks there is no differentiation needed. With `grad_method="F"`, `finite-diff` and `param_shift` will behave as expected, computing the state derivative correctly and raising an error (because param-shift+state does not work), respectively.
The only downside I currently see for `grad_method="F"` is that `finite-diff` (and `param_shift` for valid measurements, because it has a fallback logic) will create tapes to differentiate global phases even when they have no impact (i.e. expectation values, probabilities etc).

However, I'd prefer that we are wasteful (unnecessary global phases can be compiled away anyways, and then some form of caching should kick in) than returning wrong results.